docs/edgeql/select.rst

```diff
@@ -1026,6 +1026,70 @@ exclusively fetch the ``Movie.characters`` of type ``Hero``.
     ...
   }
 
+Accessing types in polymorphic queries
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While the type of an object is displayed inside polymorphic queries, this
+is simply a convenience of the REPL and not a property that can be accessed.
+This is particularly noticeable if you cast an object to ``json``, making it
+impossible to determine the type if the query is polymorphic:
+
+.. code-block:: edgeql-repl
+
+    db> select Person limit 1;
+    {default::Villain {id: 6c22bdf0-5c03-11ee-99ff-dfaea4d947ce}}
+    db> select <json>Person limit 1;
+    {Json("{\"id\": \"6c22bdf0-5c03-11ee-99ff-dfaea4d947ce\"}")}
```

Comment:
The way GitHub is diffing this makes it look a lot more different than it actually is. 😅 I'll explain what I was trying to accomplish here after the suggestion:
```suggestion
While the type of an object is displayed alongside the results of polymorphic
queries run in the REPL, this is simply a convenience of the REPL and not a
property that can be accessed. This is particularly noticeable if you cast an
object to ``json``, making it impossible to determine the type if the query is
polymorphic. First, the result of a query as the REPL presents it with type
annotations displayed:

.. code-block:: edgeql-repl

    db> select Person limit 1;
    {default::Villain {id: 6c22bdf0-5c03-11ee-99ff-dfaea4d947ce}}

Note the type ``default::Villain``, which is displayed for the user's
convenience but is not actually part of the data returned. This is the same
query with the result cast as ``json`` to show only the data returned:

.. code-block:: edgeql-repl

    db> select <json>Person limit 1;
    {Json("{\"id\": \"6c22bdf0-5c03-11ee-99ff-dfaea4d947ce\"}")}

.. note::

    We will continue to cast subesequent examples in this section to ``json``,
    not because this is required for any of the functionality being
    demonstrated, but to remove the convenience type annotations provided by
    the REPL and make it easier to see what data is actually being returned by
    the query.
```

- Clarify that the type annotation only happens in the REPL
- Split the examples with and without `json` cast to drive home the difference we're trying to call attention to
- Add a note that explains the continued use of `json` cast to any readers who might not understand and might wrongly assume you must cast to `json` to get type information back on a query.